src/goto-checker/properties.cpp

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************\
+
+Module: Properties
+
+Author: Daniel Kroening, Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Properties
+
+#include "properties.h"
+
+#include <util/invariant.h>
+
+std::string as_string(resultt result)
+{
+  switch(result)
+  {
+  case resultt::UNKNOWN:
+    return "UNKNOWN";
+  case resultt::PASS:
+    return "PASS";
+  case resultt::FAIL:
+    return "FAIL";
+  case resultt::ERROR:
+    return "ERROR";
+  }
+
+  UNREACHABLE;
+}
+
+std::string as_string(property_resultt result)
+{
+  switch(result)
+  {
+  case property_resultt::NOT_REACHED:
+    return "NOT REACHED";
+  case property_resultt::UNKNOWN:
+    return "UNKNOWN";
+  case property_resultt::NOT_REACHABLE:
+    return "UNREACHABLE";
+  case property_resultt::PASS:
+    return "PASS";
+  case property_resultt::FAIL:
+    return "FAIL";
+  case property_resultt::ERROR:
+    return "ERROR";
+  }
+
+  UNREACHABLE;
+}
+
+/// Returns the properties in the goto model
+propertiest initialize_properties(const abstract_goto_modelt &goto_model)
+{
+  propertiest properties;
+  forall_goto_functions(it, goto_model.get_goto_functions())
```

Comment:
:pick: an ranged-for would be preferred here, to make explicit that `it` is used just to access the data and not for other iterator operations. :rabbit2: